### PR TITLE
Skip --exclude-libs linker flag on macOS

### PR DESCRIPTION
**Pull request by aprocter**
The macOS linker does not seem to like the `--exclude-libs=ALL` flag used by codegen. This PR omits that flag on macOS. Not 100% sure it's the right way to handle it, but it seems to be what we are doing for `libngraph`.

### CODE REVIEW

**Comment by silee2** [comment]
<--exclude-libs> is a GNU ld only option.

File: src/ngraph/codegen/CMakeLists.txt [hunk 1]
```diff
@@ -84,9 +84,11 @@ if (NGRAPH_GPU_ENABLE OR (NGRAPH_CPU_ENABLE AND NOT NGRAPH_DEX_ONLY))
     add_dependencies(codegen header_resource)
     if (NGRAPH_CPU_ENABLE)
         add_dependencies(codegen libmkldnn libeigen)
-        # --exclude-libs=ALL prevents symbols from statically-linked libraries (LLVM, in this case)
-        # from being automatically exported
-        set_property(TARGET codegen APPEND PROPERTY LINK_FLAGS "-Wl,--exclude-libs=ALL")
+        if(NOT APPLE)
```
Comment (rkimballn1) [inline src/ngraph/codegen/CMakeLists.txt]: This won't work on WIN32

Comment (silee2) [reply src/ngraph/codegen/CMakeLists.txt]: `if(NOT LINUX)` May work.

**Comment by aprocter** [comment]
#2217 incorporated this, so closing.

**Pull request closed without merging**